src/herder/test/TransactionQueueTests.cpp

```diff
@@ -44,93 +64,69 @@ class TransactionQueueTest
-
-        if (AddResult == TransactionQueue::AddResult::ADD_STATUS_PENDING)
-        {
-            REQUIRE(afterInfo.mMaxSeq == tx->getSeqNum());
-            REQUIRE(afterInfo.mTotalFees == beforeInfo.mTotalFees + 100);
-        }
-        else
-        {
-            REQUIRE(afterInfo == beforeInfo);
-        }
     }
 
     void
-    remove(std::vector<TransactionFramePtr> const& toRemove)
+    removeAndReset(std::vector<TransactionFramePtr> const& toRemove)
     {
         auto size = mTransactionQueue.toTxSet({})->sizeTx();
-        mTransactionQueue.remove(toRemove);
-        REQUIRE(size - toRemove.size() ==
+        mTransactionQueue.removeAndReset(toRemove);
+        REQUIRE(size - toRemove.size() >=
                 mTransactionQueue.toTxSet({})->sizeTx());
     }
 
     void
-    shift(std::vector<TransactionFramePtr> const& removed = {})
+    ban(std::vector<TransactionFramePtr> const& toRemove)
     {
         auto size = mTransactionQueue.toTxSet({})->sizeTx();
-        auto removedFrom = std::map<AccountID, int>{};
-        for (auto tx : removed)
-        {
-            removedFrom[tx->getSourceID()]++;
-        }
-
-        auto beforeInfo =
-            std::map<AccountID, TransactionQueue::AccountTxQueueInfo>{};
-        for (auto from : removedFrom)
-        {
-            beforeInfo[from.first] =
-                mTransactionQueue.getAccountTransactionQueueInfo(from.first);
-            REQUIRE(beforeInfo[from.first].mMaxSeq > 0);
-        }
+        mTransactionQueue.ban(toRemove);
+        REQUIRE(size - toRemove.size() >=
+                mTransactionQueue.toTxSet({})->sizeTx());
+    }
 
+    void
+    shift()
+    {
         mTransactionQueue.shift();
-        for (auto from : removedFrom)
-        {
-            auto afterInfo =
-                mTransactionQueue.getAccountTransactionQueueInfo(from.first);
-            if (afterInfo.mMaxSeq == 0)
-            {
-                REQUIRE(afterInfo.mTotalFees == 0);
-            }
-            else
-            {
-                REQUIRE(afterInfo.mMaxSeq == beforeInfo[from.first].mMaxSeq);
-                REQUIRE(afterInfo.mTotalFees ==
-                        beforeInfo[from.first].mTotalFees - from.second * 100);
-            }
-        }
-        REQUIRE(size - removed.size() ==
-                mTransactionQueue.toTxSet({})->sizeTx());
     }
 
     void
-    check(std::vector<TransactionFramePtr> const& expected = {},
-          std::vector<TransactionFramePtr> const& banned = {})
+    check(const TransactionQueueState& state)
     {
         auto txSet = mTransactionQueue.toTxSet({});
         auto expectedTxSet = TxSetFrame{{}};
-        for (auto const& tx : expected)
+        for (auto const& accountState : state.mAccountStates)
         {
-            expectedTxSet.add(tx);
+            auto& txs = accountState.mAccountTransactions;
+            auto fees =
+                std::accumulate(std::begin(txs), std::end(txs), 0,
```

Comment:
missing include for `std::accumulate` (numeric)

Reply:
updated